charm/kafka/lib/charms/layer/kafka.py

```diff
@@ -61,8 +61,27 @@ def install(self, zk_units=[], log_dir='logs'):
 
         config = hookenv.config()
 
+        broker_id = os.environ['JUJU_UNIT_NAME'].split('/', 1)[1]
+
+        storageids = hookenv.storage_list('logs')
+        if storageids:
+            storageid = storageids[0]
+            mount = hookenv.storage_get('location', storageids[0])
+            
+            if mount:
+                broker_path = os.path.join(log_dir, '.broker_id')
+
+                if os.path.isfile(broker_path):
+                    with open(broker_path, 'r') as f:
+                        try:
+                            broker_id = int(f.read().strip())
+                        except ValueError:
+                            hookenv.log('{}'.format('invalid broker id format'))
+                            hookenv.status_set(
+                                'blocked', 'unable to validate broker id format')
```

Comment:
If we're blocking here, I think we'd want to return early. The caller might also need to halt as well.